WordPress/gutenberg
# Borders: Switch to ToolsPanel for displaying UI

Depends on:
- ~https://github.com/WordPress/gutenberg/pull/32392~

Related: 
- #36641
- https://github.com/WordPress/gutenberg/issues/27331
- https://github.com/WordPress/gutenberg/pull/33744
- https://github.com/WordPress/gutenberg/pull/34027
- https://github.com/WordPress/gutenberg/pull/34061
- https://github.com/WordPress/gutenberg/pull/36753

## Description

This PR:
- Adds new `border` InspectorControls group slot
- Updates the border block support hooks to provide callbacks required for `ToolsPanel`
- Updates the border hook to inject border support controls via the new InspectorControls slot
- Updates the global styles border panel to use the new `ToolsPanel`
- Adds default CSS styles to ensure visual borders when width, style or color are set individually

### Default Controls

_This PR only addresses the switch to the new `ToolsPanel` component. Deciding which controls should display by default for which blocks will be addressed via a separate PR._

## How has this been tested?

Manually.

#### Test instructions

1. Edit a post, add a group block, then select that block.
2. Confirm that the border controls within the sidebar are within a ToolsPanel
3. Confirm each border control works appropriately (Note: this PR does not make any specific block's border controls "default" in terms of always displaying within the new panel)
4. Switch to the site editor
5. Under "By block type", expand the group block panel and confirm the border controls are also in the new panel there
6. Test the border controls within global styles and ensure they work as expected.

## Screenshots 

https://user-images.githubusercontent.com/60436221/129139274-4dc35162-1e91-4c20-bd14-9a6cf5a91ae9.mp4

## Types of changes
New feature. Enhancement.

## Checklist:
